hsm/manager_hsm.go

```diff
@@ -51,10 +52,11 @@ var ErrPreGeneratedKeys = &fosite.RFC6749Error{
 	DescriptionField: "Cannot add/update pre generated keys on Hardware Security Module",
 }
 
-func NewKeyManager(hsm Context, config *config.DefaultProvider) *KeyManager {
+func NewKeyManager(hsm Context, ctx context.Context, config *config.DefaultProvider) *KeyManager {
 	return &KeyManager{
-		Context:      hsm,
-		KeySetPrefix: config.HSMKeySetPrefix(),
+		Context: hsm,
+		ctx:     ctx,
```

Comment:
I suggest not storing the context in the KeyManager. I'd suggest passing the configuration options (`KeySetPrefix` and `DevMode` for example) to the constructor instead and storing them directly.

Reply:
I might remember it wrong, but one of my previous code review, the suggestion was to pass configuration context so when configuration is updated, the changes will be used. Also because I am not full time go developer I code by [example quite often](https://github.com/ory/hydra/blob/v2.0.3/persistence/sql/persister.go#L106). That said I am happy to make the change - just clearifing how I got here! :)

Reply:
Hey, sorry for the late review. @alnr is correct here, the context should come from the function caller, not the service itself. I checked quickly and `getKeySetAttributes` is being called from two functions, both of which have context.Context available. I'll push the changes shortly